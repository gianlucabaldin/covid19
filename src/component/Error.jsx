import React from 'react';

const Error = () => <div>Ops! An error occured.</div>;

export default Error;
