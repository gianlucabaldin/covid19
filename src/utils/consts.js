export const DEFAUL_MAX_DATA_SIZE = 30;

export const API_ITALY_HYSTORICAL =
  'https://api.covid19api.com/total/dayone/country/italy';

export const API_ITALY_HYSTORICAL_APIFY =
  'https://api.apify.com/v2/datasets/CUdKmb25Z3HjkoDiN/items?format=json&clean=1';
export const API_ITALY_HYSTORICAL_APIFY_SHORT_URL = 'api.apify.com';

export const API_ITALY_REGION =
  'https://openpuglia.org/api/?q=getdatapccovid-19';

export const OPEN_PUGLIA_API = 'openpuglia.org';
export const COVID_19_API = 'covid19api.com';

export const STATUS = {
  CONFIRMED: 'CONFIRMED',
  RECOVERED: 'RECOVERED',
  DEATHS: 'DEATHS',
  ACTIVES: 'ACTIVES',
  INTENSIVE_THERAPY: 'INTENSIVE_THERAPY',
  TOTAL_HOSPITALIZED: 'TOTAL_HOSPITALIZED',
  TOTAL_POSITIVE: 'TOTAL_POSITIVE',
  TAMPONI: 'TAMPONI',
  TOTAL_CASES: 'TOTAL_CASES',
  DECEASED: 'DECEASED',
  NEW_DAILY_POSITIVE: 'NEW_DAILY_POSITIVE',
  DAILY_DECEASED: 'DAILY_DECEASED',
  DAILY_SWABS: 'DAILY_SWABS',
};
