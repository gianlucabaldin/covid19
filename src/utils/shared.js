/* eslint-disable import/prefer-default-export */

/**
 * Check if the value is a number and more than a thousand, in that case
 * call the toLocalString() witch inserts (eventually) the separator, used
 * in Summary
 * I.E. for italian it inserts the "." ad thousand separator
 * @param {number || string} val the value to insert the separator to if > thousand
 */
export const getLocalizedValue = (val) => {
  if (typeof val === 'number') {
    return Number(val).toLocaleString();
  }
  return val;
};
