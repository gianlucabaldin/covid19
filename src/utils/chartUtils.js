// const DEFAUL_MAX_DATA_SIZE = 20;

import { DEFAUL_MAX_DATA_SIZE } from './consts';

// In order to have a "readable" chart, reduce if too much input values
export const reduceData = (data, maxSize = DEFAUL_MAX_DATA_SIZE) => {
  const { length } = data;
  if (length > maxSize) {
    const y = Math.round(length / maxSize);
    data = data.filter((el, i) => Math.round(i % y) === 0);
  }
  return data;
};
