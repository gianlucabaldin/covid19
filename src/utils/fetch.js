import { mockItalyHistoricalResponseApify } from './mock/mockItalyHistoricalResponse_Apify';
import { mockItalyRegionsResponse } from './mock/mockItalyRegionsResponse';
import { API_ITALY_HYSTORICAL_APIFY, API_ITALY_REGION } from './consts';

export const fetchItalyHistoricalAllApify = async (mock = false) => {
  if (mock) {
    console.log('mock = true');
    return mockItalyHistoricalResponseApify;
  }
  console.log('mock = false');
  return fetch(API_ITALY_HYSTORICAL_APIFY).then((res) => res.json());
};

export const fetchItalyRegion = async (mock = false) => {
  if (mock) {
    return mockItalyRegionsResponse;
  }
  return fetch(API_ITALY_REGION).then((res) => res.json());
};
