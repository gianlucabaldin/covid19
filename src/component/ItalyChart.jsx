import React from 'react';
import '../../node_modules/react-vis/dist/style.css';
import { Container, Box } from '@material-ui/core';
import moment from 'moment';
import SwitchInterval from './SwitchInterval';
import Error from './Error';
import DataProvided from './DataProvided';
import { STATUS, API_ITALY_HYSTORICAL_APIFY_SHORT_URL } from '../utils/consts';
// import SwitchInterval from './SwitchInterval';
import Chart from './Chart';

const ItalyChart = ({
  data,
  error,
  loading,
  checked,
  onToggleSwitch,
  width = 500,
}) => {
  const {
    intensiveTherapy,
    totalHospitalized,
    totalPositive,
    newDailyPositive,
    dailyDeceased,
    dailySwabs,
  } = data;
  return (
    <>
      <Box
        marginTop={1}
        marginBottom={3}
        data-id="italy-chart-box"
        style={{
          display: 'flex',
          justifyContent: 'space-between',
          flexDirection: 'row',
          alignItems: 'center',
        }}
      >
        <DataProvided href={API_ITALY_HYSTORICAL_APIFY_SHORT_URL} />

        <SwitchInterval onToggleSwitch={onToggleSwitch} checked={checked} />
      </Box>

      <Box
        style={{
          display: 'flex',
          // justifyContent: '',
          flexDirection: 'row',
          flexWrap: 'wrap',
        }}
      >
        <Chart
          height={300}
          width={600}
          series={[intensiveTherapy]}
          status={[STATUS.INTENSIVE_THERAPY]}
        />
        <Chart
          height={300}
          width={600}
          series={[totalHospitalized]}
          status={[STATUS.TOTAL_HOSPITALIZED]}
        />
        <Chart
          height={300}
          width={600}
          series={[totalPositive]}
          status={[STATUS.TOTAL_POSITIVE]}
        />
        <Chart
          height={300}
          width={600}
          series={[newDailyPositive]}
          status={[STATUS.NEW_DAILY_POSITIVE]}
        />
        <Chart
          height={300}
          width={600}
          series={[dailyDeceased]}
          status={[STATUS.DAILY_DECEASED]}
        />
        <Chart
          height={300}
          width={600}
          series={[dailySwabs]}
          status={[STATUS.DAILY_SWABS]}
        />
      </Box>
    </>
  );
};

export default ItalyChart;
