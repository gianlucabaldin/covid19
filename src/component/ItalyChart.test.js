import React from 'react';
import { shallow } from 'enzyme';
import { XYPlot, LineMarkSeries } from 'react-vis/dist';
import ItalyChart from './ItalyChart';
import { fetchItalyHistoricalAll } from '../utils/fetch';
import Error from './Error';

describe('ItalyChart', () => {
  it('renders with data with 3 series', async () => {
    const mockData = await fetchItalyHistoricalAll(true);
    const wrapper = shallow(<ItalyChart data={mockData} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(XYPlot).length).toEqual(1);
    expect(wrapper.find(LineMarkSeries).length).toEqual(3);
  });
  it('renders en error message with no data', () => {
    const wrapper = shallow(<ItalyChart />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Error).length).toEqual(1);
  });

  it('renders error message if empty data are passed', () => {
    const wrapper = shallow(<ItalyChart data={undefined} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Error).length).toEqual(1);
  });
});
