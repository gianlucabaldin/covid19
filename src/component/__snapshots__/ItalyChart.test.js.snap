// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItalyChart renders en error message with no data 1`] = `<Error />`;

exports[`ItalyChart renders error message if empty data are passed 1`] = `<Error />`;

exports[`ItalyChart renders with data with 3 series 1`] = `
<Fragment>
  <WithStyles(ForwardRef(Grid))
    container={true}
    data-id="swith-interval-container"
    spacing={2}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <SwitchInterval />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={4}
      md={6}
      style={
        Object {
          "padding": 16,
        }
      }
      xs={12}
    >
      <Chart
        series={
          Array [
            undefined,
          ]
        }
        status={
          Array [
            "INTENSIVE_THERAPY",
          ]
        }
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={4}
      md={6}
      style={
        Object {
          "padding": 16,
        }
      }
      xs={12}
    >
      <Chart
        series={
          Array [
            undefined,
          ]
        }
        status={
          Array [
            "TOTAL_HOSPITALIZED",
          ]
        }
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={4}
      md={6}
      style={
        Object {
          "padding": 16,
        }
      }
      xs={12}
    >
      <Chart
        series={
          Array [
            undefined,
          ]
        }
        status={
          Array [
            "TOTAL_POSITIVE",
          ]
        }
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={4}
      md={6}
      style={
        Object {
          "padding": 16,
        }
      }
      xs={12}
    >
      <Chart
        series={
          Array [
            undefined,
          ]
        }
        status={
          Array [
            "NEW_DAILY_POSITIVE",
          ]
        }
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={4}
      md={6}
      style={
        Object {
          "padding": 16,
        }
      }
      xs={12}
    >
      <Chart
        series={
          Array [
            undefined,
          ]
        }
        status={
          Array [
            "DAILY_DECEASED",
          ]
        }
      />
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={4}
      md={6}
      style={
        Object {
          "padding": 16,
        }
      }
      xs={12}
    >
      <Chart
        series={
          Array [
            undefined,
          ]
        }
        status={
          Array [
            "DAILY_SWABS",
          ]
        }
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    spacing={2}
  >
    <DataProvided
      href="api.apify.com"
    />
  </WithStyles(ForwardRef(Grid))>
</Fragment>
`;
